posix.mak

```diff
@@ -601,13 +602,26 @@ style_lint: dscanner $(LIB)
 	@echo "Check that Ddoc runs without errors"
 	$(DMD) $(DFLAGS) $(NODEFAULTLIB) $(LIB) -w -D -Df/dev/null -main -c -o- $$(find etc std -type f -name '*.d') 2>&1
 
+################################################################################
+# Build dub from source if no dub is available
+################################################################################
+
+DUB_PATH=$(abspath $(shell which $(DUB)))
+ifeq (,$(DUB_PATH))
+	DUB_PATH=$(DUB_DIR)/bin/dub
+	DUB=$(DUB_DIR)/bin/dub
+endif
+
+$(DUB): | $(DUB_DIR) $(LIB)
+	cd $(DUB_DIR) && DFLAGS="$(DFLAGS) $(LIB) $(NODEFAULTLIB) $(LINKDL)" DMD=$${PWD}/$(DMD) ./build.sh
```

Comment:
Why is `$(NODEFAULTLIB)` necessary here?

Reply:
Ah, I guess because you want to use the version of druntime and phobos that were just built.

Reply:
Well, I have to build `dub` from source, because auto-tester doesn't come with a `dub` version (kinda silly) and also no globally installed dmd (`HOST_DMD` is only exposed when building dmd and my PR to fix that has remained unanswered: https://github.com/braddr/at-client/pull/9).